### DISCUSSION TITLE
email validation

### DISCUSSION THREAD

**Discussion by brunowilliang** [pocketbase/pocketbase#1333] [Q&A]
Hello, I'm passing this function to create a new user, and it doesn't arrive the email...
can someone help me?

![code3](https://user-images.githubusercontent.com/41432581/208983773-70e77542-66db-4cd2-b88e-7ec89cb3855e.png)

in my console.log i get this error
[ClientResponseError 401: The request requires valid admin authorization token to be set.]

**Comment by ganigeorgiev** [comment]
You need to call `requestVerification()` **after** creating the user.

Additionally, if you haven't already, you can check your mail settings by sending a test email from the "Admin UI > Settings > Mail settings > Send test email".

The 401 error is not from the `requestVerification()` and most likely is from some other action (it is also possible that it is from the `create` call if you haven't enabled the Create API rule for your auth collection).

**Reply by brunowilliang** [reply]
@ganigeorgiev in my API rules I didn't put anything... what can I put? to receive emails?

![Screen Shot 2022-12-21 at 16 19 59](https://user-images.githubusercontent.com/41432581/208986099-e840552f-7b3d-46b0-96dc-5ee7354f274b.png)

**Reply by ganigeorgiev** [reply]
Are you sure that the `console.log` is from any of the above 2 requests?

`requestVerification()` is always public and doesn't require authentication.

**Reply by brunowilliang** [reply]
@ganigeorgiev He gives a success, the problem is that the email doesn't arrive

**Reply by ganigeorgiev** [reply]
@Brunowilliang Have you tried sending a test email from  "Admin UI > Settings > Mail settings > Send test email"?

Also note that by default PocketBase uses the `sendmail` command to send test emails which is not reliable and a lot of mail providers (eg. gmail) will directly reject the email and refuse to deliver it.

This is why for testing purposes I recommend using a fake smtp server like [mailtrap.io](https://mailtrap.io/) (they have free plan).

For production usage you can check mailjet, sendgrid, sendinblue, mailgun, amazon ses, etc.

**Reply by ganigeorgiev** [reply]
In addition, also note that `requestVerification()` will resolve successfully even if it fail to deliver the email or a user with the provided email does not exist, to prevent users enumeration.
But in all cases you have to call `requestVerification()` **after** creating the user.

**Reply by brunowilliang** [reply]
@ganigeorgiev was solved by passing the "emailVisibility": true 👍🏻

**Reply by ganigeorgiev** [reply]
@Brunowilliang `emailVisibility` is only used as a flag to show/hide the record email when exporting the model. It is not related to the email send functionality. The issue is/was something else.
